### PR TITLE
TASK-5923 - Organization returns authentication origins secretKey

### PR DESCRIPTION
**Pull request by pfurio**

### CODE REVIEW

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/iterators/OrganizationCatalogMongoDBIterator.java [hunk 1]
```diff
@@ -105,7 +107,50 @@ private void fetchNextBatch() {
...
+
+    // TODO: Remove this method after several releases (TASK-5923)
+    /**
+     * This method should only be necessary in order to be able to run the migration for TASK-5923 because OpenCGA looks for the config.
+     * This configuration was changed in TASK-5923, therefore, OpenCGA would be unable to initialise without it.
+     * Once the migration is run, this code will do nothing, so it should be removed in the future.
+     *
+     * @param organizationDocument Organization document.
+     */
+    private void recoverSecretKeyTask5923(Document organizationDocument) {
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/iterators/OrganizationCatalogMongoDBIterator.java:125]: Once this method is deleted, the corresponding migration should be also "somehow disabled", so if it's still pending, it fails and forces you to jump to an earlier version

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/OrganizationManager.java [hunk 2]
```diff
@@ -357,4 +364,19 @@ public List<String> getOrganizationIds(String token) throws CatalogException {
         authorizationManager.checkIsOpencgaAdministrator(tokenPayload, "get all organization ids");
         return catalogDBAdaptorFactory.getOrganizationIds();
     }
+
+    private void privatizeResults(OpenCGAResult<Organization> result) {
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/OrganizationManager.java:368]: Also known as "redact"
https://www.altlaw.co.uk/blog/data-redaction-vs-data-masking-the-key-difference

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/UserManager.java [hunk 3]
```diff
@@ -590,7 +590,7 @@ public OpenCGAResult<User> update(String userId, ObjectMap parameters, QueryOpti
 
             userId = getValidUserId(userId, payload);
             for (String s : parameters.keySet()) {
-                if (!s.matches("name|email|organization|attributes")) {
+                if (!s.matches("name|email|attributes")) {
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/UserManager.java:593]: I added "this" line 10 years ago... what a horrible line...
https://github.com/opencb/opencga/commit/f51494c15d010625959c7cebaa2b5244436eea68#diff-ae76d8c01706b07219bbea60e7e513295a883a4bb9b3434a1f24897345cf4cd4R253-R257

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/UserManager.java [hunk 4]
```diff
@@ -771,7 +771,7 @@ public AuthenticationResponse login(String organizationId, String username, Stri
         auditManager.auditUser(organizationId, username, Enums.Action.LOGIN, username,
                 new AuditRecord.Status(AuditRecord.Status.Result.SUCCESS));
         String userId = authenticationFactory.getUserId(organizationId, authId, response.getToken());
-        if (!CatalogAuthenticationManager.INTERNAL.equals(authId)) {
+        if (!CatalogAuthenticationManager.OPENCGA.equals(authId) && !CatalogAuthenticationManager.INTERNAL.equals(authId)) {
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/UserManager.java:774]: Sometimes you're checking against both "opencga" and the old "internal", and some others just against "opencga"... is it justified every time?
Why not adding a method and check both always?

File: opencga-app/src/main/java/org/opencb/opencga/app/migrations/v3_0_0/OrganizationMigration.java [hunk 5]
```diff
@@ -377,17 +377,22 @@ protected void run() throws Exception {
...
                         if (authenticationOrigin.getType().equals(AuthenticationOrigin.AuthenticationType.OPENCGA)
-                                && authenticationOrigin.getId().equals(CatalogAuthenticationManager.INTERNAL)) {
+                                && "internal".equals(authenticationOrigin.getId())) {
                             continue;
                         }
-                        authenticationOrigin.setAlgorithm(algorithm);
-                        authenticationOrigin.setSecretKey(secretKey);
-                        authenticationOrigin.setExpiration(3600);
-                        authOrigins.add(convertToDocument(authenticationOrigin));
+                        Document authOriginDoc = convertToDocument(authenticationOrigin);
```
Comment (j-coll) [inline opencga-app/src/main/java/org/opencb/opencga/app/migrations/v3_0_0/OrganizationMigration.java:383]: As this super critical migration has changed, should we explicitely test this migration again?

Comment (pfurio) [reply opencga-app/src/main/java/org/opencb/opencga/app/migrations/v3_0_0/OrganizationMigration.java:383]: Yes, we should

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/UserManager.java (hunk 3 above)
Comment (pfurio) [reply opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/UserManager.java:593]: The day I completely change it you'll miss it.

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/UserManager.java (hunk 4 above)
Comment (pfurio) [reply opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/UserManager.java:774]: INTERNAL should only be checked here. I'm not checking INTERNAL anywhere else. If the installation hasn't been migrated yet and an "opencga managed" user attempts to log in, the authId for that user will still be "internal". Once it has been migrated, the auth origin for all "opencga managed" users will then become "OPENCGA".

**Review by j-coll — DISMISSED** [review]

**Pull request merged**